iothub/device/src/IAuthenticationMethod.cs

```diff
@@ -1,12 +1,14 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
     /// Authentication interface to use for device communications.
     /// </summary>
-    public interface IAuthenticationMethod
+    public interface IAuthenticationMethod : IDisposable
```

Comment:
Some of the implementations of this interface are disposable. There was no other better way to enforce disposing objects.